packages/contracts-bedrock/test/L2/L1Block.t.sol

```diff
@@ -93,3 +114,74 @@ contract L1BlockTest is CommonTest {
         });
     }
 }
+
+contract L1BlockEcotone_Test is L1BlockTest {
+    /// @dev Sets up the test suite.
+    function setUp() public virtual override {
+        super.setUp();
+
+        bytes4 functionSignature = bytes4(keccak256("setL1BlockValuesEcotone()"));
+        bytes memory callDataPacked = abi.encodePacked(
+            basefeeScalar,
+            blobBasefeeScalar,
+            sequenceNumber,
+            timestamp,
+            number,
+            basefee,
+            blobBasefee,
+            hash,
+            batcherHash
+        );
+
+        bytes memory functionCallDataPacked = abi.encodePacked(functionSignature, callDataPacked);
+
+        vm.prank(depositor);
+        (bool success, ) = address(l1Block).call(functionCallDataPacked);
+        require(success, "Function call failed");
+    }
+
```

Comment:
I think it would be more useful to have a single fuzz test that accepts values for each param and then makes an assertion on each possible param after making the call. This would be able to replace each of the individual getter tests. This will give much better test coverage. I know that this is following the same pattern as the previous tests, but also the previous tests could do the same thing

Reply:
will have to investigate how this works

Reply:
If you define a test that accepts arguments, it will be a fuzz test. You can make the arguments be the exact types that are inputs to the function and just encode them, make the call then make an assertion on each getter

Reply:
fuzz test added

Reply:
Perfect!